proposals/0018-work-graphs.md

```diff
@@ -543,9 +549,12 @@ void GroupMemoryBarrier() { Barrier(GROUP_SHARED_MEMORY, GROUP_SCOPE); }
 void GroupMemoryBarrierWithGroupSync() {
   Barrier(GROUP_SHARED_MEMORY, GROUP_SCOPE | GROUP_SYNC);
 }
-
 ```
 
+> Note: These equivalent forms are not currently mapped to the previous DXIL
+> Barrier operation on prior shader models.  The new barrier is only available
+> on SM 6.8 and above.
```

Comment:
I'm still struggling with this note. I don't think it adds clarity to the specification.

Is there a concern that the above text implied that these intrinsics would be available in older shader models that we can address by clarifying the wording elsewhere in the specification?

Reply:
There is a long-standing convention for mapping HLSL features to shader models where the compiler will map the feature if it's possible without significant complications or performance penalties.

We describe the new Barrier operations as a superset of the prior operations and list the equivalent forms to the previous HLSL intrinsics - "The pseudo-code below shows implementing the existing HLSL memory barrier functions using the new `Barrier` function.".

This describes the subset of parameter values that map directly to functionality in existing shader models.  This note is included to make it clear that even though the compiler *could* map these uses directly to existing operations on prior shader models, it does not do so.

As for alternative means of clarification, we could explicitly define something like the availability attributes on the Barrier declarations in this spec.  Is that preferable?  Even with that, one might ask *why* the equivalent forms can't be made to compile down to prior shader models, and I suppose our answer would be: because we decided not to implement that.  The note helps avoid that question, since it addresses it directly.

Reply:
Since it seems that there is some confusion over what the obvious thing to do here would be, I think some kind of a note is worthwhile.  How about:

> Note: The new barrier is only available on shader model 6.8 and above. Although there are some cases where there is an equivalent DXIL operation for prior shader models, there is not a complete mapping from one to the other, which would only make it partially available. This design allows for a consistent set of rules that can easily be validated early on in compilation.

Reply:
I've updated the note as suggested.  @llvm-beanz, please resolve if this is satisfactory.